transmart-rest-api/grails-app/controllers/org/transmartproject/rest/SystemController.groovy

```diff
@@ -55,4 +56,66 @@ class SystemController {
         }
     }
 
+    /**
+     * Clears tree node and counts caches and rebuilds the cache:
+     * <code>GET /v2/admin/system/rebuild_cache</code>
+     *
+     * This endpoint should be called after loading, deleting or updating
+     * tree nodes in the database.
+     * Only available for administrators.
+     *
+     * Asynchronous call. The call returns when rebuilding has started.
+     * Code 503 is returned if a rebuild operation is already in progress.
+     *
+     */
+    def rebuildCache() {
+        checkForUnsupportedParams(params, [])
+        if (!authContext.user.admin) {
+            throw new AccessDeniedException('Only allowed for administrators.')
+        }
+        systemResource.rebuildCache()
+        response.status = 200
+    }
+
+    /**
+     * Checks if a cache rebuild task is running:
+     * <code>GET /v2/admin/system/rebuild_status</code>
+     * Returns an object with a field <code>status</code> with value 'running'
+     * or 'stopped'.
+     * Example: <code>{"status": "running"}</code>.
+     *
+     * Only available for administrators.
+     *
+     * @return an object with a field <code>status</code> with value <code>running</code>
+     * or <code>stopped</code>.
+     *
+     */
+    def rebuildStatus() {
```

Comment:
Because you're using the same lock as for the update after data loading, it actually makes sense to have only one status call. You could populate the `updateStatus` object similarly as in that call, but then with a single task.

Reply:
Proposal here: #339.

Reply:
great, thank you! I will have a look